### ISSUE TITLE
emacs-ycmd test failures

### ISSUE THREAD

**Issue by vmandela** [NixOS/nixpkgs#26944]
## Issue description

I have been unable to get emacs-ycmd code completion to work in emacs. YouCompleteMe which uses the same underlying ycmd package works with vim. So I tried running the emacs-ycmd test suite and am seeing test failures. 

### Steps to reproduce

~~~
nix-shell -p emacs ycmd python
git clone git@github.com:abingham/emacs-ycmd.git
cd emacs-ycmd
make deps # assuming cask is already installed
make YCMDPATH=$(which ycmd) test
~~~

Test output is below

~~~
 passed  30/72  company-ycmd-test-remove-self-from-function-args
   passed  31/72  company-ycmd-test-remove-template-args-from-function-args
REQUEST [error] Error (error) while connecting to http://127.0.0.1:44123/load_extra_conf_file.
REQUEST [error] Error (error) while connecting to http://127.0.0.1:44123/event_notification.
Load YCMD extra conf /home/a0393778/p-code/emacs-ycmd/test/resources/.ycm_extra_conf.py? (y or n) n
REQUEST [error] Error (error) while connecting to http://127.0.0.1:44123/event_notification.
ERROR: No .ycm_extra_conf.py file detected, so no compile flags are available. Thus no semantic support for C/C++/ObjC/ObjC++. Go READ THE DOCS *NOW*, DON'T file a bug report.
Test flycheck-ycmd-test-error backtrace:
  flycheck-ert-should-errors((3 15 error "expected ';' at end of decla
  apply(flycheck-ert-should-errors (3 15 error "expected ';' at end of
  flycheck-ert-should-syntax-check("test-error.cpp" flycheck-ycmd-test
~~~

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...) - `17.03.1395.9b948ea439 (Gorilla)`

* Nix version: (run `nix-env --version`) - nix-env (Nix) 1.11.11
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`) - 17.03.1395.9b948ea439
* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`) - No

**Comment by cstrahan** [comment]
I don't use emacs, but try enabling logging with something like the following and report back with the results:

```elisp
(set-variable 'ycmd-server-command ("ycmd" "--keep_logfiles" "--stderr=stderr.log" "--stdout=stdout.log"))
```

Depending on whether those tests do or do not run with the same `$PWD`, you may want to provide a full path for those log files so you can easily find them (I don't know elisp, otherwise that's what I would have provided as an example). Hopefully there will be something illuminating in one of those log files.

**Comment by cstrahan** [comment]
Actually, from reading a bit further in the source:

```elisp
(defcustom ycmd-server-args '("--log=debug"
                              "--keep_logfile"
                              "--idle_suicide_seconds=10800")
  "Extra arguments to pass to the ycmd server."
  :type '(repeat string))
```

So you may want to use `ycmd-server-args` for those log related arguments that I suggested above (I don't know if it makes an material difference, though).

**Comment by vmandela** [comment]
I did some digging and found that the issue was due to proxy configuration on my system. emacs-ycmd tries to contact the ycmd server using 127.0.0.1:<port num>. The NO_PROXY environment variable on my system contained localhost but not 127.0.0.1. The communication between the ycmd server and emacs-ycmd was broken but there was no indication of this in the emacs-ycmd logs. The issue went away once I added 127.0.0.1 to the NO_PROXY variable.

This was the issue that had the pointers. 
https://github.com/abingham/emacs-ycmd/issues/404

@cstrahan  Thank you very much for your time.
